diff_drive_controller/src/velocity_limiter.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ * velocity_limiter.cpp
+ *
+ *  Created on: May 30, 2019
+ *      Author: christof
+ */
+#include <diff_drive_controller/velocity_limiter.hpp>
+#include <cassert>
+
+namespace diff_drive_controller {
+
+VelocityLimiter::VelocityLimiter()
+        : _wheel_separation(0.0f)
+        , _vel_x_max(0.0f)
+        , _mass(1.0f)
+        , _inertia(0.0f)
+        , _f_wheel_max(0.0f)
+        , _f_wheel_min(0.0f)
+        , _has_velocity_limits(false)
+        , _has_acceleration_limits(false){};
+
+bool VelocityLimiter::init(float wheel_separation, bool has_velocity_limits, bool has_acceleration_limits,
+        float vel_x_max, float vel_th_max, float acc_x_max, float acc_x_min, float acc_th_max) {
+    if (wheel_separation <= 0.0f) {
+        return false;
+    }
+    _wheel_separation = wheel_separation;
+    _has_velocity_limits = has_velocity_limits;
+    _has_acceleration_limits = has_acceleration_limits;
+
+    if (_has_acceleration_limits) {
+        if (acc_x_max <= 0 || acc_x_min >= 0) {
```

Comment:
or if min > max

Reply:
Above statement makes sure that `acc_x_max` is a positive and `acc_x_min` is a negative number. After that, it is impossible for `acc_x_min` to be greater than `acc_x_max`